packages/customize-widgets/src/components/sidebar-block-editor/use-sidebar-block-editor.js

```diff
@@ -89,12 +89,12 @@ function widgetToBlock( widget ) {
 }
 
 function initState( sidebar ) {
-	const state = { blocks: [] };
+	const state = [];
```

Comment:
The changes in this file are mainly necessary because we need to make sure different `setState` calls don't cause race condition. It's best to access the current(previous) state in the `setState` callback. This also has a benefit of not having to recreate the `onChangeBlocks` function every time state changes.

Migrating the state from an object to a plain array is just a preference of mine, and IMO makes the code a lot cleaner and "hookier".